app/battle_preparations.py

```diff
@@ -0,0 +1,25 @@
+from app.knight import Knight
+
+
+def get_knight_from_database(knight: dict) -> Knight:
+    knight_instance = Knight(
+        name=knight.get("name"),
+        power=knight.get("power"),
+        hp=knight.get("hp"),
+        armour=knight.get("armour"),
+        weapon=knight.get("weapon"),

```

Comment:
Ensure that the `weapon` attribute is correctly handled when it is `None` to avoid potential errors. Consider initializing it as an empty dictionary in the `Knight` class constructor if not already done.